lib/Partitioner/Partitioner.cpp

```diff
@@ -35,6 +42,65 @@ static uint64_t updateUsedMem(const std::set<Storage *> &usedStorage,
   return ret;
 }
 
+void Partitioner::dumpDAG(llvm::StringRef dotFilename) {
+  if (partitions_.size() == 0)
+    return;
+  std::map<BackendKind, const char *> backendName = {
```

Comment:
It seems like a bad idea to encode the names of all backends in the partitioner, because the set of backends may change over time. I think backends have a `getName()` method that you could use.

Reply:
At this stage, only "BackendKind" is kept, that is why I encoded the name. I think I will add another class member to keep the BackendName then.